docs/decisions/0018-expose-all-sass-vars-as-css-vars.rst

```diff
@@ -0,0 +1,143 @@
+advantages over SCSS variables such as:
+
+- they can be changed at runtime by changing the stylesheet, and can be
+  mainpulated using JavaScript.
+- they can be scoped to a particular CSS class or media query.
+
+SCSS/SASS still offers a number of other advantages such as nested classes,
+mixins, functions, and utilities that are currently not available with CSS.
+
+All Open edX MFEs use Bootstrap via Paragon for styling. Each MFE builds its
+own stylesheet which is a combination Bootstrap, Paragon, and some MFE-specific
+styling.
+
+Currently the SCSS code is structured as follows:
+
+- At the code we have Bootstrap
+- Then we have Paragon, which uses Bootstrap and react-bootstrap
+- Then we have the branding package which has SCSS files that can override
+  Paragon defaults and add additional SCSS code
+- Then we have MFEs that import from the branding package and Paragon, and
+  usually hav some of their own SCSS code
+
+Paragon uses Bootstrap variables, and the branding theme uses Paragon/Bootstrap
+variables, any changes to any variables mean that the entire codebase needs to
+be rebuilt. However, this need not be the case, if the dependencies between
+these packages is expressed via CSS variables instead of SCSS variables, then
+any changes to values will be picked up at runtime. Such a setup would allow
+building Bootstrap + Paragon + the branding package into a stylesheet that
+contains all the variables MFEs need so they can use the stylesheet directly
+and not need recompilation if this common stylesheet changes.
+
+This ADR only concerns itself consumers of Paragon, and as such still expects
+Paragon to continue using SCSS variables itself.
+
+If we want to be able to support a common pre-built core stylesheet that can be
+used by all MFEs and deployed independently of MFEs, then MFEs cannot use any
+SCSS variables from Paragon + Bootstrap since in that case any changes to the
+core theme will require a rebuild of each MFE, and as such these variables
+should instead be used as CSS variables so that they can be looked up at
+runtime rather then build time.
+
+Bootstrap already exports a number of CSS variables (these can be found in
+`this document for Bootstrap 4.x`_), however they don't cover all usecases. For
+instance, usage of colour variables like `primary-400`, or `light-300` are
+abundant in MFEs, however these are not expored by Bootstrap even though they
+are available as SCSS variables.
+
+Decision
+********
+
+We will export all theme variables that an MFE could need as CSS variables so
+that these can be applyed at runtime rather than build time.
+
+Paragon can create CSS variable exports like: `--pgn-blue`, `--pgn-primary`,
+`--pgn-primary-200`` etc. The reason for prefixing with `--pgn` is to isolate
+changes to these names in Bootstrap. For example, in Bootstrap 4.x these are
+named as `--blue, --primary` etc. However in `Bootstrap 5`_, they are prefixed
```

Comment:
Another reason I'd love for Paragon to move away from Bootstrap nomenclature is because consumers often rely on the internals of Bootstrap (classes, mixins, variables, etc), which in theory _should_ be just an implementation detail of the Paragon library, not really exposed to the consumer.